Histogram.py

```diff
@@ -0,0 +1,169 @@
+    Events = NanoEventsFactory.from_root(
+                file=filename,
+                    treepath="Delphes",
+                        schemaclass=DelphesSchema,
+                        ).events()
+    
+    # require two jets
+    events = Events
+    mask = ak.num(events.FatJet)>=2
+    events = events[mask]
+
+    # Dijet
+    events["Dijet"] = events.FatJet[:,0]+events.FatJet[:,1]
+
+    # transverse mass calculation
+    E1 = ET(events.Dijet)
+    E2 = events.MissingET.MET
+    MTsq = (E1+E2)**2-(events.Dijet.px+events.MissingET.px)**2-(events.Dijet.py+events.MissingET.py)
+    events["MT"] = np.sqrt(MTsq, where=MTsq>=0)
+            
+    # 4-vectors for dijet
+    events["Jet_pt"] = events.Dijet.pt
+    events["Jet_eta"] = events.Dijet.eta
+    events["Jet_phi"] = events.Dijet.phi
+    events["Jet_mass"] = events.Dijet.mass
+    # Other quantites to be plotted
+    events["MET"] = events.MissingET.MET
+    events["DeltaEta"] = events.FatJet.DeltaEta
+    events["DeltaPhi"] = events.FatJet.DeltaPhi
+
+    ## For plotting individually for jet1 and jet2
+    leading_jets = events.FatJet[:, :2]
+
+    jet1_mask = leading_jets[:, 0].pt >= leading_jets[:, 1].pt
+    jet2_mask = ~jet1_mask
+
+    events["Jet1"] = ak.where(jet1_mask, leading_jets[:, 0], leading_jets[:, 1])
+    events["Jet2"] = ak.where(jet2_mask, leading_jets[:, 0], leading_jets[:, 1])
+
+    # 4-vectors for jet1 and jet2
+    events["Jet1_pt"] = events["Jet1"].pt
+    events["Jet2_pt"] = events["Jet2"].pt
+    
+    events["Jet1_eta"] = events["Jet1"].eta
+    events["Jet2_eta"] = events["Jet2"].eta
+    
+    events["Jet1_phi"] = events["Jet1"].phi
+    events["Jet2_phi"] = events["Jet2"].phi
+    
+    events["Jet1_mass"] = events["Jet1"].mass
+    events["Jet2_mass"] = events["Jet2"].mass
+    
+    events["DeltaEta"] = np.abs(events["Jet2_eta"] - events["Jet1_eta"])
+    events["DeltaPhi"] = np.abs(normalize_angle(events["Jet1_phi"] - events["Jet2_phi"]))
+    
+    events["DeltaPhi_MET_Jet1"] = np.abs(normalize_angle(events.MissingET.phi - events["Jet1_phi"]))
+    events["DeltaPhi_MET_Jet2"] = np.abs(normalize_angle(events.MissingET.phi - events["Jet2_phi"]))
+
+    # Stable inv frac
+    dark_hadron_ids = [4900111, 4900113, 4900211, 4900213]
+    stable_particle_ids = [51, 52, 53]
+    pid = events.Particle["PID"]
+    d1 = events.Particle["D1"]
+
+    # Boolean array of whether a particle is dark
+    is_dark = ak.zeros_like(pid)
+    for dhid in dark_hadron_ids:
+        is_dark = is_dark | (np.abs(pid)==dhid)
+    is_dark = is_dark==1
+
+    # PIDs of dark daughter
+    dark_daughter = pid[d1[is_dark]] 
+    is_dark_daughter = ak.zeros_like(dark_daughter)
+
+    for dsid in stable_particle_ids:
+        is_dark_daughter = is_dark_daughter | (np.abs(dark_daughter)==dsid)
+
+    stability = [None]*len(events)
+    for i in range(len(pid)): 
+        if ak.sum(is_dark[i])!=0:
+            stability[i] = ak.sum(is_dark_daughter[i])/ak.sum(is_dark[i])
```

Comment:
you can avoid a loop by using `np.divide`:
```python
numer = ak.sum(is_dark_daughter, axis=0)
denom = ak.sum(is_dark, axis=0)
stability = np.divide(numer, denom, out=np.zeros_like(numer), where=denom>0)
```

Reply:
Awesome. I didn't know about that.

Reply:
is there a reason to revert the use of `np.divide()`?

Reply:
Yeah it was throwing errors.